server.js

```diff
@@ -372,102 +413,114 @@ app.post('*', function (req, res, next) {
+      // and send back the results to the client
+      if (asrErr) {
+        mozlog.info('request.asr.error', {
+          request_id: res.locals.request_id,
+          time: Date.now() - asr_request_start,
+        });
+        return next(asrErr);
+      }
 
-    const resBody = asrBody && asrBody.toString('utf8');
-    try {
-      res.json(JSON.parse(resBody));
-    } catch (e) {
-      mozlog.info('request.asr.error', {
+      const resBody = asrBody && asrBody.toString('utf8');
+      try {
+        res.json(JSON.parse(resBody));
+      } catch (e) {
+        mozlog.info('request.asr.error', {
+          request_id: res.locals.request_id,
+          time: Date.now() - asr_request_start,
+        });
+        return res.status(500).json({error: 'Internal STT Server Error'});
+      }
+
+      mozlog.info('request.asr.finish', {
         request_id: res.locals.request_id,
-        time: Date.now() - asr_request_start
+        status: 200,
+        time: Date.now() - asr_request_start,
       });
-      return res.status(500).json({error: 'Internal STT Server Error'});
-    }
-
-    mozlog.info('request.asr.finish', {
-      request_id: res.locals.request_id,
-      status: 200,
-      time: Date.now() - asr_request_start
-    });
 
-    if (config.s3_bucket && metadata.storetranscription === '1') {
-      const json_upload_params = {
-        Body: resBody,
-        Bucket: config.s3_bucket,
-        ContentType: 'application/json',
-        Key: key_base + '/transcript.json'
-      };
+      if (config.s3_bucket && metadata.storetranscription === '1') {
+        const json_upload_params = {
+          Body: resBody,
+          Bucket: config.s3_bucket,
+          ContentType: 'application/json',
+          Key: key_base + '/transcript.json',
+        };
 
-      const s3_request_start = Date.now();
+        const s3_request_start = Date.now();
 
-      mozlog.info('request.s3.json.start', {
-        request_id: res.locals.request_id,
-        key: key_base + '/transcript.json'
-      });
+        mozlog.info('request.s3.json.start', {
+          request_id: res.locals.request_id,
+          key: key_base + '/transcript.json',
+        });
 
-      S3.putObject(json_upload_params, (s3_error) => {
-        if (s3_error) {
-          mozlog.info('request.s3.json.error', {
+        S3.putObject(json_upload_params, (s3_error) => {
+          if (s3_error) {
+            mozlog.info('request.s3.json.error', {
+              request_id: res.locals.request_id,
+              key: key_base + '/transcript.json',
+              status: s3_error.statusCode,
+              time: Date.now() - s3_request_start,
+            });
+            return next(s3_error);
+          }
+
+          mozlog.info('request.s3.json.finish', {
             request_id: res.locals.request_id,
             key: key_base + '/transcript.json',
-            status: s3_error.statusCode,
-            time: Date.now() - s3_request_start
+            status: 200,
```

Comment:
are we sure this always return 200?

Reply:
The diff is just weird here. This maps to the old line 449, which is identical.